ts-client/src/dlmm/index.ts

```diff
@@ -828,6 +829,132 @@ export class DLMM {
     return (min ? binId.floor() : binId.ceil()).toNumber();
   }
 
+  /**
+  * The function `getLbPairLockInfo` retrieves all pair positions that has locked liquidity.
+  * @param {number} [lockDurationSecsOpt] - An optional value indicating the minimum position lock duration in seconds that the function should return. 
+  * @returns The function `getLbPairLockInfo` returns a `Promise` that resolves to a `PairLockInfo`
+  * object. The `PairLockInfo` object contains an array of `PositionLockInfo` objects.
+  */
+  public async getLbPairLockInfo(lockDurationSecsOpt?: number): Promise<PairLockInfo> {
+    const lockDurationSecs = lockDurationSecsOpt | 0;
+    const lbPairPositions = await this.program.account.positionV2.all([
+      {
+        memcmp: {
+          bytes: bs58.encode(this.pubkey.toBuffer()),
+          offset: 8,
+        },
+      },
+    ]);
+
+    // filter positions has lock_release_point > currentTimestamp + lockDurationSecs
+    const clockAccInfo = await this.program.provider.connection.getAccountInfo(SYSVAR_CLOCK_PUBKEY);
+    const onChainTimestamp = new BN(
```

Comment:
it is not always timestamp, it depends on how the pool type:
- slot -> compare with slot
- timestamp -> compare with timestamp

Reply:
i see, updated the PR with condition check for lbPair activationType